### PR TITLE
*: Remove the check of initialized auto ID

### PR DESCRIPTION
**Pull request by zimulala**
Using the way to check the initialized auto ID will have a compatibility issue with previous versions.  So we remove the check of initialized auto ID.

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @coocood @winkyao

File: meta/autoid/autoid.go [hunk 1]
```diff
@@ -179,10 +183,15 @@ func (alloc *memoryAllocator) Alloc(tableID int64) (int64, error) {
 }
 
 // NewAllocator returns a new auto increment id generator on the store.
-func NewAllocator(store kv.Storage, dbID int64) Allocator {
+func NewAllocator(store kv.Storage, originalDBID, dbID int64) Allocator {
```
Comment (hanfei1991) [inline meta/autoid/autoid.go]: Maybe we don't need to add an extra argument. If we want to use old schema ID, we just call `NewAllocator(store, oldID)`

File: meta/meta.go [hunk 2]
```diff
@@ -141,18 +141,20 @@ func (m *Meta) parseTableID(key string) (int64, error) {
 }
 
 // GenAutoTableID adds step to the auto ID of the table and returns the sum.
-func (m *Meta) GenAutoTableID(dbID int64, tableID int64, step int64) (int64, error) {
+func (m *Meta) GenAutoTableID(originalDBID, dbID, tableID, step int64) (int64, error) {
+	// Check if DB exists.
```
Comment (hanfei1991) [inline meta/meta.go]: Do we also need to check if old db exists or not ?

Comment (coocood) [reply meta/meta.go]: The old db doesn't need to exist, it can be dropped.

**Comment by hanfei1991** [comment]
LGTM

File: executor/ddl_test.go [hunk 3]
```diff
@@ -225,7 +225,7 @@ func (s *testSuite) TestRenameTable(c *C) {
 	tk.MustExec("insert rename2.t values ()")
 	tk.MustExec("rename table rename2.t to rename3.t")
 	tk.MustExec("insert rename3.t values ()")
-	tk.MustQuery("select * from rename3.t").Check(testkit.Rows("1", "2", "3"))
+	tk.MustQuery("select * from rename3.t").Check(testkit.Rows("1", "5001", "10001"))
```
Comment (coocood) [inline executor/ddl_test.go]: Why this result changed?

File: infoschema/builder.go [hunk 4]
```diff
@@ -73,7 +73,7 @@ func (b *Builder) ApplyDiff(m *meta.Meta, diff *model.SchemaDiff) ([]int64, erro
 	// We try to reuse the old allocator, so the cached auto ID can be reused.
 	var alloc autoid.Allocator
 	if tableIDIsValid(oldTableID) {
-		if oldTableID == newTableID {
+		if oldTableID == newTableID && diff.Type != model.ActionRenameTable {
```
Comment (coocood) [inline infoschema/builder.go]: We can still reuse the allocator for rename table, just add a method `SetDBID`.

File: executor/ddl_test.go (hunk 3 above)
Comment (zimulala) [reply executor/ddl_test.go]: Because of the change at your second question.

File: infoschema/builder.go (hunk 4 above)
Comment (zimulala) [reply infoschema/builder.go]: But, it needs to add an interface to Allocator.

Comment (coocood) [reply infoschema/builder.go]: Alright, skip auto ID when rename table is acceptable.

**Pull request merged**